app/controllers/services_controller.rb

```
@@ -13,20 +13,32 @@ class ServicesController < ApplicationController
 
   def index
     # use SFSG as default site
-    # TODO: be able to parse both categories and eligibilities at once
-    if params[:category_id]
-      all_services = find_by_category(params[:category_id], @site_id_string)
-    elsif params[:eligibility_id]
-      all_services = find_by_eligibility(params[:eligibility_id], @site_id_string)
-    end
+    matching_services = find_services(
+      params[:category_id],
+      params[:eligibility_id],
+      @site_id_string
+    )
 
-    render json: ServicesWithResourceLitePresenter.present(all_services)
+    render json: ServicesWithResourceLitePresenter.present(matching_services)
   end
 
   def read_site_id_string
     @site_id_string = params[:site_id] || Site.find_by(site_code: 'sfsg').id.to_s
   end
 
+  def find_services(category_id, eligibility_id, site_id)
+    if category_id == 'all' || eligibility_id == 'all'
+      find_all(site_id)
+    # TODO: be able to parse both categories and eligibilities at once
+    elsif category_id
+      find_by_category(category_id, site_id)
+    elsif eligibility_id
+      find_by_eligibility(eligibility_id, site_id)
+    else
+      []
```

Should we return an error code if expected data is not being passed in?